cardshowviewtakenpicturesview/src/main/java/br/com/stant/libraries/cardshowviewtakenpicturesview/CardShowTakenPictureViewImagesAdapter.java

```diff
@@ -2,54 +2,77 @@
 
 import android.databinding.DataBindingUtil;
 import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.helper.ItemTouchHelper;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 
+import org.jetbrains.annotations.NotNull;
+
 import java.util.ArrayList;
 import java.util.List;
 
 import br.com.stant.libraries.cardshowviewtakenpicturesview.databinding.CardShowTakenPictureViewImageRecycleItemBinding;
+import br.com.stant.libraries.cardshowviewtakenpicturesview.domain.enums.CardShowTakenPictureStateEnum;
 import br.com.stant.libraries.cardshowviewtakenpicturesview.domain.model.CardShowTakenImage;
+import br.com.stant.libraries.cardshowviewtakenpicturesview.utils.VibratorUtils;
+import br.com.stant.libraries.cardshowviewtakenpicturesview.utils.listener.DragAndDropHandler;
+import br.com.stant.libraries.cardshowviewtakenpicturesview.utils.listener.ItemTouchHelperViewHolder;
 import io.reactivex.Observable;
 
 import static br.com.stant.libraries.cardshowviewtakenpicturesview.utils.ImageDecoder.setImageBitmapToImageView;
 
-public class CardShowTakenPictureViewImagesAdapter extends RecyclerView.Adapter<CardShowTakenPictureViewImagesAdapter.ItemViewHolder> {
+public class CardShowTakenPictureViewImagesAdapter extends RecyclerView.Adapter<CardShowTakenPictureViewImagesAdapter.ItemViewHolder>
+        implements DragAndDropHandler {
 
+    private CardShowTakenPictureView mView;
     private List<CardShowTakenImage> mCurrentCardShowTakenImageList;
     private List<CardShowTakenImage> mOriginalTempCardShowTakenImageList;
     private List<CardShowTakenImage> mCardShowTakenImageListAsAdded;
     private List<CardShowTakenImage> mCardShowTakenImageListAsRemoved;
-    private CardShowTakenPictureView mView;
+    private ItemTouchHelper mItemTouchHelper;
 
-    public CardShowTakenPictureViewImagesAdapter(List<CardShowTakenImage> imageUrlsList, CardShowTakenPictureView view) {
-        this.mCurrentCardShowTakenImageList      = imageUrlsList;
+    public CardShowTakenPictureViewImagesAdapter(CardShowTakenPictureView view) {
         this.mView                               = view;
-        this.mOriginalTempCardShowTakenImageList = new ArrayList<>();
-        this.mCardShowTakenImageListAsAdded      = new ArrayList<>();
-        this.mCardShowTakenImageListAsRemoved    = new ArrayList<>();
+        this.mCurrentCardShowTakenImageList      = new ArrayList<>(0);
```

Comment:
remove "this" if has "m" pattern on global variables

Reply:
Fixed